src/discovery/Discovery.ts

```diff
@@ -113,12 +113,12 @@ class Discovery {
         // The sigchain data of the vertex (containing all cryptolinks)
         let vertexChainData: ChainData = {};
         // If the vertex we've found is our own node, we simply get our own chain
-        const nodeId = nodesUtils.decodeNodeId(vertexGId.nodeId);
+        const nodeId = nodesUtils.decodeNodeId(vertexGId.nodeId)!;
```

Comment:
Initially I thought that we couldn't make this `!` assertion here - see my original comment from #318 on this: https://github.com/MatrixAI/js-polykey/pull/318#discussion_r793295864

However, we can, purely because we should be doing some verification on the `vertex` **before** we add it to the `vertexQueue` (where we retrieved this vertex from). Making a comment on that now.

Reply:
Yea, SoT is expected to be correct. No need to do double-validation, validation on input and validation on output, that's just a waste of compute.

Reply:
The verification of the `vertex` will be handled on the boundary (i.e. where we actually receive the claim data - `requestChainData`